### PR TITLE
feat(backup): remote backup for all subscriptions

### PR DESCRIPTION
**Pull request by gsanchietti**
Now both Enterprise and Community subscriptions have access to the remote backup.

Requires: https://github.com/NethServer/nethsecurity/pull/652

Exposed also some more info:
- creation date in local time
- encrypted/not encrypted flag
- size

Also exposed the delete backup button.

Backup list:
image

Restore list:
image

Delete confirmation:
image

NethServer/nethsecurity#664

### CODE REVIEW

File: public/i18n/en/translation.json [hunk 1]
```diff
@@ -575,7 +575,15 @@
...
         "date": "Date",
-        "backups": "Backups"
+        "backups": "Backups",
+        "octet-stream": "Encrypted",
+        "gzip": "Not encrypted",
+        "pgp-encrypted": "Encrypted",
+        "size": "Size",
+        "mimetype": "Type",
+        "delete": "Delete",
+        "modal_delete_title": "Delete remote backup",
```
Comment (andre8244) [inline public/i18n/en/translation.json:585]:
```suggestion
        "modal_delete_title": "Delete backup",
```

Comment (gsanchietti) [reply public/i18n/en/translation.json:585]: Why removing the "delete" world?

File: src/components/standalone/backup_and_restore/ModalDeleteBackup.vue [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+<template>
+  <NeModal
+    :primary-button-disabled="loadingdelete"
+    :primary-button-loading="loadingdelete"
+    :primary-label="t('standalone.backup_and_restore.backup.delete')"
+    :secondary-button-disabled="loadingdelete"
+    :title="t('standalone.backup_and_restore.backup.modal_delete_title')"
+    :visible="showDeleteModal"
+    kind="info"
```
[the remark this answers is not in the document]
Comment (gsanchietti) [reply src/components/standalone/backup_and_restore/ModalDeleteBackup.vue:79]: Not sure this should be a warning: the action is not catastrophic.

[the remark this answers is not in the document]
Comment (gsanchietti) [reply src/components/standalone/backup_and_restore/ModalDeleteBackup.vue]: Done but not really convinced.

File: src/components/standalone/backup_and_restore/ModalDeleteBackup.vue [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    primary-button-kind="primary"
```
[the remark this answers is not in the document]
Comment (gsanchietti) [reply src/components/standalone/backup_and_restore/ModalDeleteBackup.vue]: Done

File: src/components/standalone/backup_and_restore/ModalDeleteBackup.vue [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+const emit = defineEmits(['close'])
+
+let loadingdelete = ref(false)
+
+let objNotification = {
+  notificationTitle: '',
+  notificationDescription: ''
+}
+
+let errordeleteBackup = ref({ ...objNotification })
```
[the remark this answers is not in the document]
Comment (gsanchietti) [reply src/components/standalone/backup_and_restore/ModalDeleteBackup.vue]: Done.

File: src/components/standalone/backup_and_restore/BackupContent.vue [hunk 5]
```diff
@@ -18,32 +18,48 @@ import {
...
 interface Backup {
-  file: string
+  id: string
   name: string
+  created: BigInteger
+  size: BigInteger
+  mimetype: string
+}
+
+interface BackupDelete {
```
[the remark this answers is not in the document]
Comment (gsanchietti) [reply src/components/standalone/backup_and_restore/BackupContent.vue]: Removed: I was fighting against trypescript

File: src/components/standalone/backup_and_restore/ModalDeleteBackup.vue [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  },
+  selectedBackupLabel: {
+    type: String,
+    required: true
+  }
+})
+
+const emit = defineEmits(['close'])
+
+let loadingdelete = ref(false)
```
[the remark this answers is not in the document]
Comment (gsanchietti) [reply src/components/standalone/backup_and_restore/ModalDeleteBackup.vue]: Done

File: src/components/standalone/backup_and_restore/ModalDeleteBackup.vue [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+        id: props.selectedBackupId
+      }
+    }
+
+    let res = await ubusCall('ns.backup', methodCall, payload)
+    if (res?.data?.result === 'success') {
+      emit('close')
+    }
+  } catch (exception: any) {
+    errordeleteBackup.value.notificationTitle = t('error.cannot_delete_backup')
```
[the remark this answers is not in the document]
Comment (gsanchietti) [reply src/components/standalone/backup_and_restore/ModalDeleteBackup.vue]: Done.

File: public/i18n/en/translation.json (hunk 1 above)
Comment (andre8244) [reply public/i18n/en/translation.json:586]: This is the only place where we use the term "remote"; if we want to focus on this concept I think we should have written "remote" in other places too :man_shrugging: We can leave it as it is anyway, it's not so important

Comment (gsanchietti) [reply public/i18n/en/translation.json:586]: I agree, we should add the "remote" word in other labels

**Pull request merged**